backend/fms_core/viewsets/project.py

```diff
@@ -59,4 +62,26 @@ def summary(self, _request):
             "total_count": Project.objects.count(),
             "open_count": Project.objects.filter(status="Open").count(),
             "closed_count": Project.objects.filter(status="Closed").count(),
-        })
\ No newline at end of file
+        })
+    
+    @action(detail=False, methods=["post"])
+    def add_samples_to_study(self, request, pk=None):
+        sample_ids = request.data.get("sample_ids")
+        project_id = request.data.get("project_id")
+        study_letter = request.data.get("study_letter")
+        step_order = request.data.get("step_order", None)
+
+        samples =  make_generator(Sample.objects.filter(id__in=sample_ids))
```

Comment:
don't think you need make_generator here as much as you like it ...

Reply:
i think you might need to call ".all()" after your ".filter()" to iterate through the returned objects.